src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py

```diff
@@ -467,3 +476,55 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
                  .format(database_engine, resource_group, replicas[0]), checks=NoneCheck())
         self.cmd('{} flexible-server delete -g {} --name {} --yes'
                  .format(database_engine, resource_group, replicas[1]), checks=NoneCheck())
+
+
+class FlexibleServerLocalContextScenarioTest(LocalContextScenarioTest):
+
+    postgres_location = 'eastus'
+    mysql_location = 'westus2'
+
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=postgres_location)
+    def test_postgres_flexible_server_local_context(self, resource_group):
+        self._test_flexible_server_local_context('postgres', resource_group)
+
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=mysql_location)
+    def test_mysql_flexible_server_local_context(self, resource_group):
+        self._test_flexible_server_local_context('mysql', resource_group)
+
+    def _test_flexible_server_local_context(self, database_engine, resource_group):
+        from knack.util import CLIError
+
+        if not self.cli_ctx.local_context.is_on:
+            self.cmd('local-context on')
+
+        server_name = self.create_random_name(SERVER_NAME_PREFIX, SERVER_NAME_MAX_LENGTH)
+        if database_engine == 'postgres':
+            location = self.postgres_location
+        elif database_engine == 'mysql':
+            location = self.mysql_location
+
+        self.cli_ctx.local_context.set(['all'], 'resource_group_name', resource_group)
+        self.cli_ctx.local_context.set(['all'], 'location', location)
+
+        self.cmd('{} flexible-server create -n {}'.format(database_engine, server_name))
+
+        self.cmd('{} flexible-server show'.format(database_engine))
+
+        self.cmd('{} flexible-server update --backup-retention {}'
+                 .format(database_engine, 10))
+
+        self.cmd('{} flexible-server stop'.format(database_engine))
+
+        self.cmd('{} flexible-server start'.format(database_engine))
+
+        self.cmd('{} flexible-server restart'.format(database_engine))
+
+        self.cmd('{} flexible-server list'.format(database_engine))
+
+        self.cmd('{} flexible-server show-connection-string'.format(database_engine))
+
+        self.cmd('{} flexible-server list-skus'.format(database_engine))
+
+        self.cmd('{} flexible-server delete --yes'.format(database_engine))
```

Comment:
could you turn off local context when all operations are completed?

Reply:
@arrownj for awareness.

Reply:
Actually the base class `LocalContextScenarioTest` will automatically turn on/off local context in the `setUp` and `tearDown` step. I'm wondering why we need to manually turn it on here ?

Reply:
I didn't know it will be automatically turned off so manually turned on just in case. Thanks for letting me know

Reply:
Thanks @arrownj! @DaeunYim In this way, you don't need to change any more because you are using `LocalContextScenarioTest`